inference-engine/include/ie_api.h

```diff
@@ -104,6 +107,15 @@
  * @param type A plugin type
  */
 
+#if (__GNUC__ >= 4)  // NOLINT
```

Comment:
I think we need `#if (__GNUC__ >= 4)` -> `#if defined(__GNUC__  ) && (__GNUC__ >= 4)`